syntax = "proto3";

package jiaa.tracking;

option java_multiple_files = true;
option java_package = "com.jiaa.common.tracking";

service TrackingService {
  // 음성 스트리밍 -> STT 변환
  rpc TranscribeAudio(stream AudioRequest) returns (AudioResponse);
  
  // 실행 중인 앱 목록 전송 (변경 시에만)
  rpc SendAppList(AppListRequest) returns (AppListResponse);
}

// ==================== Audio ====================

message AudioRequest {
  bytes audio_data = 1;       // VAD로 걸러진 음성 바이너리
  bool is_final = 2;          // True = "문장 끝났음(침묵 1초 감지)"
  int64 timestamp = 3;        // 로깅용 시간
  string media_info_json = 4; // 미디어 정보 JSON
}

message AudioResponse {
  string transcript = 1;      // 변환된 텍스트
  bool is_emergency = 2;      // 위급 상황 여부
  string intent = 3;          // 의도 분석 결과
}

// ==================== App List ====================

message AppListRequest {
  // 실행 중인 앱 목록 JSON
  // Format: [{"name": "Chrome", "title": "YouTube", "url": "...", "is_active": true}]
  string apps_json = 1;
  int64 timestamp = 2;        // 전송 시간
  string clipboard_payload = 3; // AES Encrypted Payload (Base64)
  string clipboard_key = 4;     // RSA Encrypted AES Key (Base64)
}

message AppListResponse {
  bool success = 1;           // 수신 성공 여부
  string message = 2;         // 응답 메시지
  string command = 3;         // 명령 (e.g. "KILL")
  string target_app = 4;      // 대상 앱 (e.g. "Minecraft")
}
